Define environment requirements for Moodle 5.1

HuongNV13 opened the pull request:
More details:
https://moodle.atlassian.net/browse/MDL-85362

mickhawkins reviewed (commented):
Hi @HuongNV13,

Thanks for working on this, the change you've made looks correct, I'm just wondering whether we need to update any of these:

1. [Changelog](https://github.com/moodlehq/moodle-plugin-ci/blob/b2e64bb0890ed7fb3bb7b5e04af3efdce5eb6fc1/docs/CHANGELOG.md?plain=1) (and potentially minor version bump?)
2. [.github/workflows/test.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/b2e64bb0890ed7fb3bb7b5e04af3efdce5eb6fc1/.github/workflows/test.yml#L65)
3. The Travis configs ([.travis.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/b2e64bb0890ed7fb3bb7b5e04af3efdce5eb6fc1/.travis.yml#L6) or [.travis.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/b2e64bb0890ed7fb3bb7b5e04af3efdce5eb6fc1/.travis.dist.yml#L7)) (not sure if these matter since they're going away).

HuongNV13 commented:
Thanks, @mickhawkins 
I have added new lines to the changelog.

.github/workflows/test.yml is designed to test selftest for this repo, but yes, I have modified it to Postgres 15

According to Travis CI, it has been deprecated in 4.5.8 and will be removed in 5.0.0.

Cheers!

mickhawkins reviewed (approved):
Thanks @HuongNV13, this looks good to me.
